### PR TITLE
CASMNET-1636 addition of UAN hardening role for UAN repo preventing ssh to management nodes

### PR DESCRIPTION
**Pull request by alexanderkingh**
CASMNET-1636 addition of UAN hardening role for UAN repo preventing ssh to management nodes. Adds optional role for customers to run to control blocking ssh out NMN network from UANs to management nodes.

## Summary and Scope

Adds a new uan_hardening role that allows customer to optionally include blocking ssh out of UANs to NCN Management nodes over the NMN network.

## Issues and Related PRs

_List and characterize relationship to Jira/Github issues and other pull requests. Be sure to list dependencies._

* Resolves [CASMNET-1636](https://jira-pro.its.hpecorp.net:8443/browse/CASMNET-1636)
* Change will also be needed in `CASMNET-1636`

* Documentation changes required in [issue id](issue link)

### Tested on:

  * `vale`
  * Local development environment
  * Virtual Shasta

### Test description:

_How were the changes tested and success verified? If schema changes were part of this change, how were those handled in your upgrade/downgrade testing?_

- Were the install/upgrade-based validation checks/tests run (goss tests/install-validation doc)?
- Were continuous integration tests run? If not, why?
- Was upgrade tested? If not, why?
- Was downgrade tested? If not, why?
- Were new tests (or test issues/Jiras) created for this change?

## Risks and Mitigations

_Are there known issues with these changes? Any other special considerations?_

## Pull Request Checklist

- [ ] Version number(s) incremented, if applicable
- [ ] Copyrights updated
- [ ] License file intact
- [ ] Target branch correct
- [ ] CHANGELOG.md updated
- [ ] Testing is appropriate and complete, if applicable
- [ ] [HPC Product Announcement](https://cray.slack.com/archives/C026TVCSXLH) prepared, if applicable

### CODE REVIEW

**Comment by alanm-hpe** [comment]
Please update the copyrights to 2022. You'll also need to increment a patch version in a few places. See this PR for an example:
https://github.com/Cray-HPE/uan/pull/37/files

File: ansible/roles/uan_hardening/defaults/main.yml [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+#
+# Defaults for uan_hardening role
+
+# DO NOT SET VALUES IN THIS FILE.
+# Override these values in the Ansible inventory or in a playbook.
+
+# See the role README.md file for information about these variables.
+
+---
+disable_ssh_out_nmn_to_management_ncns: no
```
Comment (alanm-hpe) [inline ansible/roles/uan_hardening/defaults/main.yml:32]: With this not being included in the rest of the role `uan_access_control`, I think we can enable this by default.

Comment (alexanderkingh) [reply ansible/roles/uan_hardening/defaults/main.yml:32]: Done

**Comment by alexanderkingh** [comment]
Not sure why the continuous-integration/jenkins/pr-merge and continuous-integration/jenkins/branch failed. it does build and works for me in vale and ansible does not give this error. Screenshots in jira and output there show that as well. Maybe something with build environment for uan.

File: ansible/roles/uan_hardening/tasks/disable_ssh_out_nmn_to_management_ncns.yml [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+# UAN interface, and only for destination TCP port 22 ssh traffic.
+# Does not affect inbound traffic.
+- name: Block ssh to NCN IPs from UAN NMN$
+  iptables:
+    chain: OUTPUT
+    jump: DROP
+    protocol: tcp
+    out_interface: nmn0
+    destination_port: 22
+    destination: "{{ nmn_network_cidr }}"
```
Comment (johren-hpe) [inline ansible/roles/uan_hardening/tasks/disable_ssh_out_nmn_to_management_ncns.yml:47]: You will also need to block the NMNLB network.   An SSH to any of those IPs will take you to one of the worker nodes.

For example:
```
uan01:~ # ssh root@10.92.100.71
Password: 
Last login: Mon Jun 27 00:42:53 2022 from 10.252.1.21
ncn-w002:~ # 
{noformat}

**Review by johren-hpe — CHANGES_REQUESTED** [review]

File: ansible/roles/uan_hardening/tasks/disable_ssh_out_nmn_to_nmn_lb.yml [hunk 3]
```diff
@@ -0,0 +1,58 @@
+#
```
Comment (alanm-hpe) [inline ansible/roles/uan_hardening/tasks/disable_ssh_out_nmn_to_nmn_lb.yml:1]: Was this file included by mistake? There is a similarly named one that is included in the main.yml tasks.

File: ansible/roles/uan_hardening/tasks/disable_ssh_out_uan_to_nmn_lb.yml [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+
+    - name: "Set NMNLB subnet details"
+      set_fact:
+        nmnlb_subnet_details: "{{ item.ExtraProperties.Subnets }}"
+      loop: "{{ sls_nmnlb.json }}"
+
+    - name: "Set NMNLB subnet facts from SLS API retrieved details"
+      set_fact:
+        nmnlb_network_cidr: "{{ item.CIDR }}"
+      loop: "{{ nmnlb_subnet_details }}"
```
Comment (alanm-hpe) [inline ansible/roles/uan_hardening/tasks/disable_ssh_out_uan_to_nmn_lb.yml:45]: Is the loop necessary? Could this be simplified to:
```
    - name: "Set NMNLB subnet facts from SLS API retrieved details"
      set_fact:
        nmnlb_network_cidr: "{{ sls_nmnlb.json.ExtraProperties.Subnets.CIDR }}"
```

**Comment by alexanderkingh** [comment]
Conflicts were not resolving for some reason.

Made a new PR here New PR and branch for the uan hardening for 1.2.6 requesting your review on https://github.com/Cray-HPE/uan/pull/39. Thank you. Had around circular merge conflicts so made new pr and abandoning old one.

Please review new PR with all the changes and when reviewed i'll abandon this pr and branch

**Pull request closed without merging**